src/main/java/org/opensearchmetrics/metrics/release/CodeCoverage.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearchmetrics.metrics.release;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.http.util.EntityUtils;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.opensearchmetrics.model.codecov.CodeCovResponse;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.Optional;
+
+public class CodeCoverage {
+    private final CloseableHttpClient httpClient;
+
+    @Inject
+    public CodeCoverage() {
+        this(HttpClients.createDefault());
+    }
+
+    @VisibleForTesting
+    CodeCoverage(CloseableHttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    public CodeCovResponse coverage(String branch, String repo) {
+        String codeCovRepoURL = String.format("https://api.codecov.io/api/v2/github/opensearch-project/repos/%s/commits?branch=%s", repo, branch);
```

Comment:
So we are monitoring at the branch level for now I believe?

Reply:
Yes the release branch coming from https://github.com/opensearch-project/opensearch-metrics/blob/main/src/main/java/org/opensearchmetrics/metrics/release/ReleaseInputs.java.